### PR TITLE
feture: Styling updates to paragraph and text items (M2-7458)

### PR DESCRIPTION
**Pull request by ramirlm**
🔗 [Jira Ticket M2-7458](https://mindlogger.atlassian.net/browse/M2-7458)

Design updates to both paragraph and text items, including:
- Border radius
- Error handling
- Error state styling

### 📸 Screenshots
CleanShot 2024-09-03 at 18 00 37

CleanShot 2024-09-03 at 18 03 04

### CODE REVIEW

File: src/entities/activity/ui/items/TextItem.tsx [hunk 1]
```diff
@@ -24,5 +35,24 @@ export const TextItem = ({ item, value, onValueChange, isDisabled }: TextItemPro
...
+      />
+      <Box
+        display="flex"
+        justifyContent="flex-end"
+        alignItems="center"
+        fontSize="small"
+        color={hasError ? `${Theme.colors.light.error}` : Theme.colors.light.outline}
+        mr={2}
+      >
+        {`${t('charactersCount', { numCharacters, maxCharacters: maxResponseLength })}`}
```
Comment (felipeMetaLab) [inline src/entities/activity/ui/items/TextItem.tsx:54]: force it as string is it necessary ? only { t(...)} would it return a string ?

File: src/shared/ui/Items/Text/index.tsx [hunk 2]
```diff
@@ -35,18 +26,22 @@ export const TextItem = ({
...
                   width: '4px',
                 },
                 '&::-webkit-scrollbar-thumb': {
                   backgroundColor: '#C2C7CF',
-                  borderRadius: '100px',
                 },
               },
               '& .MuiInputBase-root': {
-                paddingRight: '2px', // Remove padding inside the input
+                paddingRight: '2px', // Reduce padding inside the input
```
Comment (felipeMetaLab) [inline src/shared/ui/Items/Text/index.tsx:38]: consider remove comments.

File: src/entities/activity/ui/items/TextItem.tsx [hunk 3]
```diff
@@ -11,10 +17,15 @@ type TextItemProps = {
...
+  const [hasError, setHasError] = useState(false);
+
+  const numCharacters = value?.length || 0;
+
+  const { t } = useCustomTranslation();
 
   const onHandleValueChange = (value: string) => {
     if (value.length > maxResponseLength) {
-      return;
+      setHasError(true);
```
Comment (ChaconC) [inline src/entities/activity/ui/items/TextItem.tsx:28]: `setHasError` should happen regardless of the value (similar to ParagraphText)

CleanShot 2024-09-03 at 10 58 24@2x

File: src/entities/activity/ui/items/TextItem.tsx [hunk 4]
```diff
@@ -11,10 +17,15 @@ type TextItemProps = {
 
 export const TextItem = ({ item, value, onValueChange, isDisabled }: TextItemProps) => {
   const { maxResponseLength } = item.config;
+  const [hasError, setHasError] = useState(false);
```
Comment (ChaconC) [inline src/entities/activity/ui/items/TextItem.tsx:20]: Should be initialized with the proper value
CleanShot 2024-09-03 at 10 57 52@2x

**Review by ChaconC — CHANGES_REQUESTED** [review]

**Review by felipeMetaLab — COMMENTED** [review]
consider these 2 reviews

File: src/entities/activity/ui/items/TextItem.tsx (hunk 4 above)
Comment (ChaconC) [reply src/entities/activity/ui/items/TextItem.tsx]: It's showing 304/300 (error) in a default color variant (meaning no error). This was after I moved to the next question and then back

File: src/widgets/Survey/model/validateItem.ts [hunk 5]
```diff
@@ -29,6 +29,17 @@ function isAnswerShouldBeNumeric(item: appletModel.ItemRecord) {
...
 
+function isAnswerTooLarge(item: appletModel.ItemRecord) {
+  if (
+    !['paragraphText', 'text'].includes(item.responseType) ||
+    !('maxResponseLength' in item.config) ||
+    typeof item.answer[0] !== 'string'
+  ) {
+    return false;
+  }
+  return item.answer[0].length > item.config.maxResponseLength ?? true;
```
Comment (ChaconC) [inline src/widgets/Survey/model/validateItem.ts:40]: Do you need this fallback value?

File: src/i18n/en/translation.json [hunk 6]
```diff
@@ -19,6 +19,9 @@
     "pleaseProvideAdditionalText": "Please provide additional text",
     "pleaseListenToAudio": "Please listen to the audio until the end.",
     "onlyNumbersAllowed": "Only numbers are allowed",
+    "answerTooLarge": "Please provide a shorter answer",
+    "question_count_plural": "{{length}} questions",
+    "question_count_singular": "{{length}} question",
```
Comment (felipeMetaLab) [inline src/i18n/en/translation.json:24]: I didn't find these question_count_singular, question_count_plural on fr translation, is it right ?

Comment (felipeMetaLab) [reply src/i18n/en/translation.json:24]: oh I see, so question_count_plural and question_count_singular is not relevant to be added on fr Translation ?
I understand its a point that you only resolved the conflict, but I didnt find this on fr Translation anyway, if it is not relevant ok, we can proceed without add those variable to fr translation.

Comment (felipeMetaLab) [reply src/i18n/en/translation.json:24]: no haha, I suggested to add on fr Translation, but I see, ok if you dont think its relevant, for me that's ok, I agree with Carlos review and you can resolve my review.

**Review by ChaconC — APPROVED** [review]

File: src/i18n/en/translation.json (hunk 6 above)
Comment (ChaconC) [reply src/i18n/en/translation.json:24]: Good catch @felipeMetaLab 

@ramirlm this is a side effect from your merge from dev into this branch. These lines were removed in [this PR](https://github.com/ChildMindInstitute/mindlogger-web-refactor/pull/530/files#diff-748406503f0dd61aa7860bd36e02a6674aac6247b7e9bbf07dd262966fbaeff0) so they shouldn't be added back in here.

**Comment by ChaconC** [comment]
Please review failed lint and test checks @ramirlm

**Review by ChaconC — APPROVED** [review]

**Pull request merged**